### ISSUE TITLE
Feature request: torch.isclose should set default atol and rtol based on the dtype of the tensors it's given

### ISSUE THREAD

**Issue by justinblaber** [pytorch/pytorch#41651]
I have a feeling `torch.isclose`'s tolerances should scale depending on dtype. For instance, if dtype is `float64`, then `atol` should be `1e-8`, but if dtype is `float32` then `atol` should maybe be `1e-6`, and similarly for `float16`. 

Thoughts?

cc @mruberry @rgommers @ezyang @SsnL @gchanan

**Comment by justinblaber** [comment]
Basically if you do any arithmetic on `float32`, it will rarely hit that `1e-8` atol tolerance, so the defaults are almost useless for the default type (unlike `np.isclose` where the default type is usually float64).

**Comment by gchanan** [comment]
I suspect we don't want to diverge from numpy semantics here, but @mruberry would know more.

**Comment by mruberry** [comment]
I updated the title of this issue. 

This is a neat idea, @justinblaber, and we actually do something like this in our test suite to facilitate comparisons across different dtypes. There are a few issues, though, that probably make it impractical to pursue:

- It's BC-breaking, as @gchanan mentions, and BC-breaking changes are painful.
- It's easy to work around by writing your own wrapper for `torch.isclose` (like our test suite does), so the value is low.
- The default absolute and relative tolerances are ALREADY absurd for some comparisons. For example, if working with values very close to zero a default absolute tolerance of 1e-8 could make everything appear equal. So users already have to be (at least a little) thoughtful about how they set them. 
- It would also requiring coordinating with NumPy so they would also break BC, which they have been loathe to do for `isclose` in the past, or we'd be divergent from NumPy, which could be confusing to users.

To summarize: it is a neat idea, and in practice I can imagine a lot of developers writing a wrapper to do this, just like we do in our test suite. But because of the challenges listed and the ease of implementing a workaround I don't think this is something we're looking to invest in.

Does that make sense?

**Comment by justinblaber** [comment]
@mruberry 

Sounds good to me. I appreciate the comprehensive response.
